shell/common/shell_test_platform_view_metal.mm

```diff
@@ -8,64 +8,50 @@
 
 #include <utility>
 
-#include "flutter/fml/platform/darwin/scoped_nsobject.h"
 #include "flutter/shell/gpu/gpu_surface_metal_impeller.h"
 #include "flutter/shell/gpu/gpu_surface_metal_skia.h"
 #include "flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.h"
 #include "flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalSkia.h"
 
+FLUTTER_ASSERT_ARC
+
 namespace flutter {
 namespace testing {
 
-static fml::scoped_nsprotocol<id<MTLTexture>> CreateOffscreenTexture(id<MTLDevice> device) {
+// This is out of the header so that shell_test_platform_view_metal.h can be included in
+// non-Objective-C TUs.
+struct DarwinContextMetal {
+  FlutterDarwinContextMetalSkia* skia_context;
+  FlutterDarwinContextMetalImpeller* impeller_context;
```

Comment:
Could this have been a union or std::variant? Yes. But we're going to delete the skia backend eventually anyway, so may as well avoid the extra complexity and make life easy on the person who deletes it.